accepted/0000-buildpack-set-env-vars-defaults.md

```diff
@@ -0,0 +1,215 @@
+user can therefore configure their build with this option using:
+```
+pack build ruby-app --env BUNDLE_DISABLE_CHECKSUM_VALIDATION=true
+```
+which **will** successfully disable checksum validation during the build phase.
+The environment variable will not be set at launch time.
+
+#### Behaviour After Proposed Change
+With the proposed change, users would be able to set the build-time value of
+this environment variable exactly as before. The proposed change would **not**
+set `BUNDLE_DISABLE_CHECKSUM_VALIDATION=true` at launch time, since no
+buildpack directly manipulates this environment variable.
+
+To change the launch-time value, the user could inject a new value at container
+run time:
+```
+docker run ruby-app --env BUNDLE_DISABLE_CHECKSUM_VALIDATION=true
+```
+
+Alternately, they could use the Environment Variables buildpack to bake a
+different launch-time default into the image:
+```
+pack build node-app --env BUNDLE_DISABLE_CHECKSUM_VALIDATION=true \
+                    --env BPE_DEFAULT_BUNDLE_DISABLE_CHECKSUM_VALIDATION=true
+```
+In this case, the build- and launch-time values of
+`BUNDLE_DISABLE_CHECKSUM_VALIDATION` is `true`.
+
+### Example: `JAVA_TOOL_OPTIONS`
+#### Current Behaviour
+Currently, the `JAVA_TOOL_OPTIONS` environment variable is set at launch time
+by various buildpacks including the Java [Debug
+Buildpack](https://github.com/paketo-buildpacks/debug/blob/ebc132acf0ed8bd084839263d37a0a8c9846e41c/helper/debug.go#L59)
+and the [JMX
+Buildpack](https://github.com/paketo-buildpacks/jmx/blob/5d2c6097bc439a0717dbbda330bdee4a32db42d5/helper/jmx.go#L44).
+Users can influence the values set by the buildpack by specifying the
+[buildpack-specific environment variables (e.g.
+`BPL_JVM_HEAD_ROOM`)](https://paketo.io/docs/buildpacks/language-family-buildpacks/java/#configuring-jvm-at-runtime)
+at launch time. The buildpacks compute correct values to populate the
+`JAVA_TOOL_OPTIONS` based on the values of the other environment variables and
+certain launch-time configuration (e.g.  memory available to the container at
+run time).
+
+If a user provides a value of `JAVA_TOOL_OPTIONS` at build time, its value will
+be available to all of the buildpacks at build time, but the inputted value of
+the environment variable **is not** added to the launch-time environment.
+
+If a user sets `JAVA_TOOL_OPTIONS` at launch time, the [user-provided flags are
+appended to the buildpack-calculated
+ones](https://paketo.io/docs/buildpacks/language-family-buildpacks/java/#configuring-jvm-at-runtime).
+
+#### Behaviour After Proposed Change
+With the proposed change, if a user specifies
+```
+pack build my-java-image --env JAVA_TOOL_OPTIONS='-a -b'
+```
+
+Then at build time,  `JAVA_TOOL_OPTIONS='-a -b'` and at launch time,
+`JAVA_TOOL_OPTIONS='-a -b <other flags as determined by buildpack(s)>'` by
+default. Note that the value provided at build time is prepended to
+buildpack-calculated set of launch-time flags.
```

Comment:
I don't think this is going to work for Java developers. It is common for users to configure the size of different memory during build with values that are not appropriate for launch.